projects/plugins/jetpack/modules/widget-visibility/widget-conditions.php

```diff
@@ -10,23 +10,22 @@ class Jetpack_Widget_Conditions {
 	static $passed_template_redirect = false;
 
 	public static function init() {
-		if ( is_admin() ) {
+		global $pagenow;
+
+		if ( is_customize_preview() || 'widgets.php' === $pagenow || 'themes.php' === $pagenow ||
+			 str_starts_with( $_SERVER['REQUEST_URI'], '/wp-json/wp/v2/widget-types' ) ||	// Widget editing via API in gutenberg widgets
+			 str_starts_with($_SERVER['REQUEST_URI'], '/wp-json/batch/v1' )) {	// Saving widgets via API in gutenberg widgets
```

Comment:
I think that `/wp-json/batch/v1` is not specific to widgets. Can we limit it to specific paths?

https://make.wordpress.org/core/2020/11/20/rest-api-batch-framework-in-wordpress-5-6/

Also, you will eventually need to address all PHPCS warnings.

Reply:
I was concerned about that being too vague, I'll take a deeper look.

> eventually need to address all PHPCS warnings.

Will do, it's complaining about str_starts_with being php8 only, the symfony php8 polyfill package is included but it's not a direct dependency. I'll just refactor to avoid str_starts_with.

Reply:
> I think that /wp-json/batch/v1 is not specific to widgets. Can we limit it to specific paths?

I'm struggling a bit with this @cpapazoglou - currently the init method is hooked into the init action, which happens before basically everything - the request hasn't yet been parsed from the HTTP request. I don't think that our init method should be responsible for trying to parse a batch request body - that's the wrong place for it, so I think the correct thing is to hook into a different (later) action.

Perhaps [rest_request_before_callbacks](https://developer.wordpress.org/reference/hooks/rest_request_before_callbacks/) would be a good choice - however it would only work on REST requests, so we'd need to keep our existing hook.

Reply:
As we discussed in the call, replicating `add_filter( 'widget_update_callback', array( __CLASS__, 'widget_update' ), 10, 3 );` inside a condition that runs for all `/wp-json/batch/v1` requests should be safe cause `widget_update_callback` is pretty specific.